archunit/src/test/java/com/tngtech/archunit/core/importer/ClassFileImporterGenericMethodSignaturesTest.java

```diff
@@ -0,0 +1,1016 @@
+                List.class, Serializable.class, Map.class, String.class);
+    }
+
+    @Test
+    @UseDataProvider("data_of_imports_complex_type_with_multiple_nested_parameters_with_generic_array_bounds")
+    public void imports_complex_type_with_multiple_nested_parameters_with_generic_array_bounds(JavaCodeUnit codeUnit) {
+        assertThatCodeUnit(codeUnit)
+                .hasTypeParameter("A")
+                .withBoundsMatching(
+                        parameterizedType(List.class).withTypeArguments(
+                                genericArray(typeVariableArrayName("X", 1)).withComponentType(typeVariable("X").withUpperBounds(Serializable.class))))
+                .hasTypeParameter("B")
+                .withBoundsMatching(
+                        parameterizedType(List.class).withWildcardTypeParameterWithUpperBound(
+                                genericArray(typeVariableArrayName("X", 2)).withComponentType(
+                                        genericArray(typeVariableArrayName("X", 1)).withComponentType(
+                                                typeVariable("X").withUpperBounds(Serializable.class)))))
+                .hasTypeParameter("C")
+                .withBoundsMatching(
+                        parameterizedType(Map.class).withTypeArguments(
+                                wildcardType().withLowerBound(
+                                        genericArray(typeVariableArrayName("Y", 1)).withComponentType(
+                                                typeVariable("Y").withUpperBounds(String.class))),
+                                parameterizedType(Map.class).withTypeArguments(
+                                        parameterizedType(Map.class).withTypeArguments(
+                                                wildcardType().withLowerBound(
+                                                        genericArray(typeVariableArrayName("Y", 3)).withComponentType(
+                                                                genericArray(typeVariableArrayName("Y", 2)).withComponentType(
+                                                                        genericArray(typeVariableArrayName("Y", 1)).withComponentType(
+                                                                                typeVariable("Y").withUpperBounds(String.class))))),
+                                                wildcardType()),
+                                        genericArray(typeVariableArrayName("X", 2)).withComponentType(
+                                                genericArray(typeVariableArrayName("X", 1)).withComponentType(
+                                                        typeVariable("X").withUpperBounds(Serializable.class)))))
+                );
+    }
+
+    private static Object[][] genericSignatureDataPointsFromConstructorAndMethod(Class<?> typeParametersOnConstructor, Class<?> typeParametersOnMethod, Class<?>... additionalImports) {
```

Comment:
Yeah, but "class with constructor" is not really the point and I would argue that every class has some constructor :wink: Also "constructor and method" tells me too little about the use case :thinking: 
What I wanted to express was basically "we always have the two parallel test cases of generic signature on constructor and generic signature on method -> this method takes always pairs of these two test cases and creates data points from them" :thinking: 

Okay, so this is what I came up with:
The method is called `testCasesFromSameGenericSignatureOnConstructorAndMethod(..)` to make it clear what the use case is (i.e. run the same tests against a constructor and a method signature).
The classes are called `GenericSignatureOn{Constructor/Method}`, because that is the point (at least for me :wink:)
I hope it's an okay compromise!